Sources/MapboxMaps/Gestures/GestureHandlers/PinchGestureHandler.swift

```diff
@@ -1,63 +1,70 @@
     }
 
-    @objc internal func handlePinch(_ pinchGestureRecognizer: UIPinchGestureRecognizer) {
-
-        delegate.cancelGestureTransitions()
-        let pinchCenterPoint = pinchGestureRecognizer.location(in: view)
-
-        if pinchGestureRecognizer.state == .began {
-
-            self.previousScale = 1.0
-            delegate.gestureBegan(for: .pinch)
+    @objc private func handleGesture(_ gestureRecognizer: UIPinchGestureRecognizer) {
+        guard let view = gestureRecognizer.view else {
+            return
+        }
 
-            self.initialCameraState = self.delegate.cameraState()
-            self.initialPinchCenterPoint = pinchCenterPoint
+        let pinchCenterPoint = gestureRecognizer.location(in: view)
 
-            /**
-             TODO: Handle a concurrent rotate gesture here.
-             Prioritize the correct gesture by comparing the velocity of competing gestures.
-             */
-        } else if pinchGestureRecognizer.state == .changed {
-            if pinchGestureRecognizer.numberOfTouches < 2 {
+        switch gestureRecognizer.state {
+        case .began:
+            initialPinchMidpoint = pinchCenterPoint
+            initialCameraState = mapboxMap.cameraState
+            cameraAnimationsManager.cancelAnimations()
+            delegate?.gestureBegan(for: .pinch)
+        case .changed:
+            guard gestureRecognizer.numberOfTouches == 2,
+                  let initialCameraState = initialCameraState,
+                  let initialPinchCenterPoint = initialPinchMidpoint else {
                 return
             }
+            cameraAnimationsManager.cancelAnimations()
+
+            let zoomIncrement = log2(gestureRecognizer.scale)
+            var cameraOptions = CameraOptions()
+            cameraOptions.center = initialCameraState.center
+            cameraOptions.padding = initialCameraState.padding
+            cameraOptions.zoom = initialCameraState.zoom
 
-            let zoomIncrement = log2(pinchGestureRecognizer.scale)
-            delegate.pinchChanged(withZoomIncrement: zoomIncrement,
-                                  targetAnchor: pinchCenterPoint,
-                                  initialAnchor: self.initialPinchCenterPoint,
-                                  initialCameraState: self.initialCameraState)
+            mapboxMap.setCamera(to: cameraOptions)
 
-            previousScale = pinchGestureRecognizer.scale
-        } else if pinchGestureRecognizer.state == .ended
-            || pinchGestureRecognizer.state == .cancelled {
+            mapboxMap.dragStart(for: initialPinchCenterPoint)
+            let dragOptions = mapboxMap.dragCameraOptions(
+                from: initialPinchCenterPoint,
+                to: pinchCenterPoint)
+            mapboxMap.setCamera(to: dragOptions)
+            mapboxMap.dragEnd()
 
-            delegate.pinchEnded()
+            mapboxMap.setCamera(
+                to: CameraOptions(
```

Comment:
nit: indentation feels weird here. 
nit: make camera options on a different line

Reply:
I think this indentation is advantageous for a few reasons…

* It keeps the lines from getting too long because Xcode's indention behaviors only increase by 4 spaces per level when you indent this way.
* It helps to avoid needing to reindent subsequent lines if, for example, you rename `setCamera` or `CameraOptions` down the road. (unlikely in this situation, but as a general rule, this is useful)
* It provides a natural visual hierarchy of nested invocations and their associated parameters.

Reply:
Is this a style that we are choosing to follow in our codebase? I see the arguments for it, but IMO, I think this style of indentation is harder to read and follow. It doesn't feel swifty. 

I think we should standardize these things in our coding style guide

Reply:
> I think we should standardize these things in our coding style guide

Agreed. We don't have a standard here, so right now it's anyone's opinion.